service/src/main/java/uk/nhs/adaptors/gp2gp/common/service/ConfidentialityService.java

```diff
@@ -38,4 +40,12 @@ private boolean hasNOPATMetaSecurity(Resource resource) {
             .stream()
             .anyMatch(this::isNOPATCoding);
     }
+
+    public boolean hasNOPATSecurityLabel(Resource resource) {
+        return (resource instanceof DocumentReference documentReference)
+               && documentReference
+                   .getSecurityLabel()
+                   .stream()
+                   .anyMatch(codeableConcept -> codeableConcept.getCoding().stream().anyMatch(this::isNOPATCoding));
+    }
```

Comment:
Can we modify this method please to make use of generics?

I think we can achieve this via a signature like:

```java
public <R extends DocumentReference> boolean hasNOPATSecurityLabel(R reference) {
    return reference.getSecurityLabel()
                   .stream()
                   .anyMatch(codeableConcept -> codeableConcept.getCoding().stream().anyMatch(this::isNOPATCoding));
}
```

Reply:
I think it is good to keep it as non-generatic as in this case I need to specifically know that it is DocumentReference and not any other type. For example, When_GenerateAndIsRedactionMessageAndNonNOPATMetaSecurityIsPresent_Expect_EmptyOptional test passes ReferralRequest into generateConfidentialityCode method hence `hasNOPATSecurityLabel` method would fail if we specifically will be looking only for DocumentReference.